src/value_serializer.rs

```diff
@@ -0,0 +1,109 @@
+use crate::Context;
+use crate::HandleScope;
+use crate::Isolate;
+use crate::Local;
+
+use crate::support::MaybeBool;
+
+use crate::data::ArrayBuffer;
+use crate::data::Value;
+
+use std::ffi::c_void;
+use std::mem::MaybeUninit;
+
+/// Instances of this class can be passed
+// Must be >= sizeof(v8::ValueSerializer), see v8__ValueSerializer__CONSTRUCT().
+#[repr(C)]
+pub struct ValueSerializer([usize; 1]);
+
+extern "C" {
+  fn v8__ValueSerializer__CONSTRUCT(
+    buf: *mut MaybeUninit<ValueSerializer>,
+    isolate: *mut Isolate,
+  );
+  fn v8__ValueSerializer__DESTRUCT(this: *mut ValueSerializer);
+
+  fn v8__ValueSerializer__TransferArrayBuffer(
+    this: *mut ValueSerializer,
+    transfer_id: u32,
+    array_buffer: Local<ArrayBuffer>,
+  );
+
+  fn v8__ValueSerializer__WriteHeader(this: *mut ValueSerializer);
+  fn v8__ValueSerializer__WriteValue(
+    this: *mut ValueSerializer,
+    context: Local<Context>,
+    value: Local<Value>,
+  ) -> MaybeBool;
+  fn v8__ValueSerializer__WriteUint32(this: *mut ValueSerializer, value: u32);
+  fn v8__ValueSerializer__WriteUint64(this: *mut ValueSerializer, value: u64);
+  fn v8__ValueSerializer__WriteDouble(this: *mut ValueSerializer, value: f64);
+  fn v8__ValueSerializer__WriteRawBytes(
+    this: *mut ValueSerializer,
+    source: *const c_void,
+    length: usize,
+  );
+}
+
+impl ValueSerializer {
+  pub fn new<'s>(scope: &mut HandleScope<'s>) -> Self {
```

Comment:
As this function does not return a Local or anything else that borrows from `scope`, I think you don't need a named lifetime here. Instead:

```rust
pub fn new(scope: &mut HandleScope<'_>) -> Self
```

...should work.